### PR TITLE
Added a link to a recent video covering emberjs testing strategies at…

### PR DESCRIPTION
**Pull request by grandschemathings**
… Yahoo

### CODE REVIEW

**Comment by mixonic** [comment]
@trek please +1 the idea of "additional resources" with links to video. I already merged one, I can clean it up if you think they are more a distraction than helpful.

**Comment by grandschemathings** [comment]
I only added the videos because I found there was already one guide page (source/routing/link-to-router-flow.md) with a youtube link. I'm just getting into ember and am trying to only include video content from 2015 that is single topic focused and of high quality. If you give the go ahead I'll work on making these all consistent as I read through the guide. @trek has to approve this one though - the speaker mentions two of his libraries for API testing :)

**Comment by michaelrkn** [comment]
@trek @locks do either of you have any feelings about this?

**Comment by trek** [comment]
So, we've traditionally avoided linking to specific third party resources for a a few reasons:

* We don't want to open the "but why didn't you include _my_ blog post" flood gates
* The more external resources we link the larger the review burden becomes in each revision to ensure that all materials still represent current best practices
* Existing community sources of these materials (e.g. http://emberwatch.com/) do a better job of collecting and curating these resources than we can

@mixonic I've added this to the agenda for Friday.

**Comment by michaelrkn** [comment]
I tend to agree with @trek's points, fwiw.

**Comment by trek** [comment]
Core team meeting has been postponed until 2015-08-03

**Comment by grandschemathings** [comment]
I'm going to respectfully disagree that "e.g. http://emberwatch.com/" does a better job.

There are 6 categories on emberwatch:

Talks: has only 3 from 2015, one titled TBA ;) .. 

Screencasts: All of the 2015 contributions go to emberscreencasts.com .. the most recent one before that was a YEAR older. The ones going to emberscreencasts are free with a leadin to 'pro' subscription for more info.

Podcasts: Fairly active in 2015 from a variety of sources, but I wasn't proposing using those.

Tutorials: Nothing after Jul 2014

Books: 12 of them, but no indication what version of ember they relate to or release dates

Cookbook: Seems like an idea waiting for good content.


So having looked that over, let me give you my perspective and ideas coming as a newbie into emberjs and javascript frameworks in general.

It's incredibly difficult to associate resources to the current version of emberjs, when I asked around people said one of the biggest issues with ember in the past was it seemed like it was changing all the time so they went with angular or w/e. My impression is the framework has settled in and won't be going through such turbulence for awhile(?) and has a chance to gain real market/mindshare.

The hurdle will always be the fact that there are a lot of resources out there telling you the OLD way to do things, which is probably a moving target.

As far as provenance of content, I think a safe direction to go is - only materials presented at meetups and embercons, that ensures it isn't a commercial slant, private project, and has been vetted to some extent by the moderators of the meetup.

The better meetups seem to have video and slides available, there are generally detailed notes about what the meetup will cover, and you actually know when the meetup occurred.

I don't want to put endless links to other resources, there can be a grab-bag at the end of the guide, but I've seen some high quality presentations looking at the various ember meetup channels on youtube and think the community would benefit from having a 30-60 minute demonstration of a concept covered in the guide, with slides when available.

tl;dr:
limit 'Additional Resources' links on any guide page to meetup and con content to prevent overload of submissions, commercialism, me-too-ism. Quality will be better because speakers are vetted and typically good quality recordings. Meetups will be inspired to do make better recordings and presentations for consideration for guide.

**Comment by locks** [comment]
@grandschemathings https://github.com/emberwatch/emberwatch

**Comment by michaelrkn** [comment]
I'm inclined to agree with @trek on this, FWIW, especially the point on having to audit all of the resources linked to before every release. Instead of linking to outside resources that currently do a better job than the Guides, let's improve the Guides.

**Comment by grandschemathings** [comment]
Final comment from me - even if the guides are improved, it doesn't address the reality that there are different learning styles. If the format of the guides were the absolute best way to convey information, people wouldn't bother with meetups, they'd just send around PDFs and save the commute. I think inclusion of some high quality video resources will make learning and adoption much easier.

e.g.
EMBERCONF 2015 - TEST-DRIVEN DEVELOPMENT BY EXAMPLE by Toran is night and day compared to reading the guide. Really contextualizes the workflow, it's like pair programming and getting a good vim/tmux tutorial all in one.

The majority of the talks that would even be useful are from EmberConf or from meetups in SV/SF/NYC/Philly and maybe a few others and most of those talks are from core team members anyway :)

@locks the issue with updating emberwatch is there's no one-one correlation between those talk titles and the guides.

**Comment by michaelrkn** [comment]
> there are different learning styles

Agreed. I believe the plan is to link to to sites that curate outside resources from http://emberjs.com/community/.

**Comment by locks** [comment]
> @locks the issue with updating emberwatch is there's no one-one correlation between those talk titles and the guides.

Are you saying the talk titles don't match guide sections, or that emberwatch doesn't have the metadata?
The latter can be fixed with PRs to the project, the former sounds like a reason to not link to talks from the guides.

> I think inclusion of some high quality video resources will make learning and adoption much easier.

What happens when a video uses deprecated features? Do we remove the link?
Who will be responsible for reviewing all the linked material every release?
What if the video owner removes the video, or the link turns otherwise dead? Do we have to patch any guide that mentions it? (major PITA at the moment)

> Final comment from me - even if the guides are improved, it doesn't address the reality that there are different learning styles.

Exactly, hence why we rely on the community for alternative methods through blog posts, screencasts, presentations, etc. I agree that discoverability of community content is important, I just don't think it should be a responsibility of the guides. There are several ideas to improve emberwatch, I'm certain they're open to more ideas on how to do that, especially if you are able to lend a hand.
I really think emberwatch can be the emberobserver of community content.

And to reiterate how little maintenance overhead we can currently afford, we removed the cookbook (crude move to emberwatch) and JSBins (no support for ES6 modules) and we're still playing catch up with Ember's releases.

**Comment by grandschemathings** [comment]
@ 
>Are you saying the talk titles don't match guide sections, or that emberwatch doesn't have the metadata?
>The latter can be fixed with PRs to the project, the former sounds like a reason to not link to talks from the guides.

It's precisely the opposite.  For example core team member Stefan Penner gave a talk at Silicon Valley Ember Meetup on June 15 (that's pretty fresh content) called "The observer tip-jar" He didn't call it "The object model -> Observers" to match the guide, but it's precisely the kind of content that would supplement the guide brilliantly.

>I think inclusion of some high quality video resources will make learning and adoption much easier.
>What happens when a video uses deprecated features? Do we remove the link?

If the features are deprecated the guide will be getting updated anyway so I don't see that as a major objection. But you (I, anyone) would either delete it or comment on it and add a video from a meetup where someone described the new feature.

>Who will be responsible for reviewing all the linked material every release?

Just doing some basic math here. Let's say there are 5 active meetups with quality content, they meet once a month. That makes only 5 new videos to even potentially consider per month. I'll just guess and say a dozen annually from a conference.

>What if the video owner removes the video, or the link turns otherwise dead? Do we have to patch any guide that mentions it? (major PITA at the moment)

It will be unlikely, I believe, that for example, NYC or Silicon Valley Ember meetups would randomly delete their youtube channel.

>Final comment from me - even if the guides are improved, it doesn't address the reality that there are different learning styles.
>Exactly, hence why we rely on the community for alternative methods through blog posts, screencasts, presentations, etc. I agree that discoverability of community content is important, I just don't think it should be a responsibility of the guides. There are several ideas to improve emberwatch, I'm certain they're open to more ideas on how to do that, especially if you are able to lend a hand.
>I really think emberwatch can be the emberobserver of community content.

>And to reiterate how little maintenance overhead we can currently afford, we removed the cookbook (crude move to emberwatch) and JSBins (no support for ES6 modules) and we're still playing catch up with Ember's releases.

Again, if the rules are 'only videos from meetups and confs' then the upper limit for pull requests in a given month would be 5, and the odds are the people authorized to approve them are already aware of them anyway.

**Comment by locks** [comment]
> If the features are deprecated the guide will be getting updated anyway so I don't see that as a major objection.

You misunderstand. For example, as soon as routable components land, every resource that uses routable controllers will have to be removed. Or when the new modules land.

> Just doing some basic math here. Let's say there are 5 active meetups with quality content, they meet once a month. That makes only 5 new videos to even potentially consider per month. I'll just guess and say a dozen annually from a conference.

There are more than 5 active meetups, each with multiple presentations. There's at least two big Ember.js conferences a year (three when Ember Fest happens), and Ember presentations at random conferences, plus screencasts.

But let's say there's only 5 videos a month. Release cycles are 6 weeks, so we're bound to catch two meetups, which means approximately 10 person-hours to review them for inclusion, plus review any currently linked resource.

> It will be unlikely, I believe, that for example, NYC or Silicon Valley Ember meetups would randomly delete their youtube channel.

Stranger things have happened, especially due to DMCAs.

I think you're underestimating the maintenance cost, I even forgot to mention in my previous post that the tutorial was also removed because of it. Core has final say, so let's wait for the meeting.

**Comment by wifelette** [comment]
@grandschemathings I think a lot of your points are correct. That said, even if we sort out that the commitment is _less_ intense than @locks is painting it to be, your ideas still mean a commitment to additional workload. 

I'm unsure how aware the public is about our efforts, but a bunch of us, including @locks, have been working hard over the last month to reinvest in docs work (some of which has already landed, much of which is still in progress). Those who stepped up have been _great_, but there are still not enough people around and helping to manage even what we consider to be the bare minimum for solid, up-to-date docs. 

I say this for context. I want to make sure that it's clear that what you're hitting here isn't just generic stop-energy, or NIH-idea-rejection. It's a very practical "no one we already know has time to really take this on" consideration.

I wonder if there's a better solution, where we keep linking to EmberWatch as the main idea, but find new people to reinvest in _that_. I don't know much about the folks running it, does anyone here? If it's not well maintained, can it be transitioned more towards being it's _own_ OSS project, and cultivating some contributors focused there?

Either way, regardless of the specific path (EmberWatch or something new or owned by Ember), we can't  really get much done on _any_ of these fronts unless there are specific people stepping up to help make it happen, and stick around to maintain at least in the immediate future.

@grandschemathings —is that perhaps you? 
Is it anyone else here who isn't already working on the docs themselves?

**Comment by grandschemathings** [comment]
@wifelette How can I help? Would it be better to email you?

**Comment by locks** [comment]
@grandschemathings you can also join us at #documentation on Slack (join via https://ember-community-slackin.herokuapp.com/ ) and we can start sorting it out :)

**Comment by trek** [comment]
The core team consensus was that we don't have the bandwidth to also get into the business of curating links, but are amenable to posting topic-specific links to other curated lists.

So, we'd be fine with a link to something like http://emberwowsocool.info/topics/testing, but not http://someguynamedjoe.it/my-blog-post-about-tests
